### ISSUE TITLE
check for timpi in config

### ISSUE THREAD

**Issue by BalticPinguin** [libMesh/libmesh#2412]
I tried to install a fresh repo using
```
../dev_tests/configure --prefix=/data/tm162/bin/${ARCH}/test_libmesh-mpich/ \
   --enable-default-comm-world \
   --disable-strict-lgpl     \
   --enable-reference-counting \
   --disable-maintainer-mode \
   --enable-qhull          \
   --enable-petsc          \
   --enable-mpi            \
   --enable-complex        \
   --enable-tetgen         \
   --enable-ifem           \
   --enable-everything     >& configs.conf
```
and forgot to make the ``git submodule update --init`` before (I suspect this to be the issue?)
Resulting, my config sais (compared to the established branch) that metaphysicl is missing(hinting me to the submodule update) and thus configures without.
But ``make`` than fails due to missing ``timpi``.
I guess, this is missing due to missing submodule update?

If ``timpi`` is required, it might be nice to check for its existence during configure and exit if not found?

**Comment by jwpeterson** [comment]
> If timpi is required, it might be nice to check for its existence during configure and exit if not found?

I agree, we should have added such a configure test after timpi was moved to a submodule... I will try and look into adding this soon.

**Comment by roystgnr** [comment]
I wish there was a way to set a git submodule to automatically update.  MetaPhysicL is optional and probably always will be, so I guess there's no need to auto-download it, but TIMPI is a 100% mandatory dependency and unless someday it gets popular enough for us to support third-party TIMPI installs we're always going to want to clone and use our contrib submodule there.

**Comment by jwpeterson** [comment]
Fixed in #2412.

**Comment by BalticPinguin** [comment]
great, thank you very much!
